aten/src/ATen/cuda/CUDAFuture.h

```diff
@@ -142,6 +142,16 @@ struct TORCH_CUDA_CPP_API CUDAFuture : at::ivalue::Future {
     return data_ptrs;
   }
 
+  bool completed() const override {
+    TORCH_WARN_ONCE(
+        "Calling completed() on instance of at::CUDAFuture. Note that this "
+        "does not have the same semantics as completed() on a vanilla "
+        "at::ivalue::future, as this returns true if the CUDA operation has "
+        "enqueued on the stream and makes no guarantees regarding actual "
+        "execution or stream synchronization.");
+    return at::ivalue::Future::completed();
```

Comment:
Slightly prefer `super()::` to `at::ivalue::Future::`, in case the parent class is changed in the future.

Reply:
I've never seen `super()` being used in C++ (unlike Python), could you point me to some docs about this?